dask_sql/physical/utils/sort.py

```diff
@@ -12,6 +17,22 @@ def apply_sort(
     sort_ascending: List[bool],
     sort_null_first: List[bool],
 ) -> dd.DataFrame:
+    # Try fast path for multi-column sorting before falling back to
+    # sort_partition_func.  Tools like dask-cudf have a limited but fast
+    # multi-column sort implementation.  We check if any sorting/null sorting
+    # is required.  If so, we fall back to default sorting implementation
+    if (
+        dask_cudf is not None
+        and isinstance(df, dask_cudf.DataFrame)
+        and all(sort_ascending)
+        and not any(sort_null_first)
+    ):
+        try:
+            df = df.sort_values(sort_columns, ignore_index=True)
+            return df.persist()
```

Comment:
1. We should not call `.persist()` on single patition frames . 

2. Just curious , Does  `.persist()` ensure  we dont trigger duplicate computations as IIRC, `.sort_values()` is not lazy. 

I wonder if this is a better patten 

```python
 df = df.persist()
 df = df.sort_values(sort_columns, ignore_index=True).persist()
```

Reply:
1. Agreed, I can add in a call to `map_partitions` in the single partition case.
2. @quasiben might know better than me the implications of calling `persist` here; I would assume this is here mostly to match up with the persist call happening in the workaround:

https://github.com/rapidsai/dask-sql/blob/4d5f7dd4c133f7f2c584db2492c8f0ef3cba7e31/dask_sql/physical/utils/sort.py#L38

Reply:
> Just saw your edit - knowing that, it looks like the current pattern should be good (once we account for the single partition case) - should we still opt to persist before running sort_values?

Testing it again now., will update here. Sorry for the edit and confusion.

Reply:
So i tested an example workflow with and without persisting first, and persisting before sorting indeed prevents duplicate computation.  





**Without Persisting ([DASK PROFILE](https://gist.githack.com/VibhuJawa/b137bdc9585a6231f85d59f22726be3c/raw/2a8084e0381e34a557cb02acc3504ca0d113186c/sort-without-persist.html)):** 
```python
st = time.time()
with performance_report(filename="sort-without-persist.html"):
    df =  dask_cudf.read_parquet(get_fp("web_sales"),columns= columns).shuffle(['ws_sold_date_sk','ws_ship_date_sk'])
    df = df.sort_values(by=['ws_bill_cdemo_sk'],ignore_index=True).persist()
    df = wait(df);
    del df
print(f"et -st = {et-st}")
```

```
et -st = 23.0989 
```

With Persisting ([DASK PROFILE](https://gist.githack.com/VibhuJawa/ca327afed58f7d1186aa839a9f5304f6/raw/799213cd123095c22b0514617ce80925aa37889a/sort-with-persist.html)): 
```python
st = time.time()
with performance_report(filename="sort-with-persist.html"):
    df =  dask_cudf.read_parquet(get_fp("web_sales"),columns= columns).shuffle(['ws_sold_date_sk','ws_ship_date_sk'])
    df = df.persist().sort_values(by=['ws_bill_cdemo_sk'],ignore_index=True).persist()
    df = wait(df);
    del df
    
et = time.time()
print(f"et -st = {et-st}")
```
```
et -st = 16.24
```

The trade of here is memory vs duplicate computation. I think we might want to think more about this .  

I wonder if a version of in-place sorting might prevent some memory overheads. 


Anyways, we should think deeply about this.

Reply:
CC: @randerzander

Reply:
I suspect the persist calls here are due to handling the multi-col sort on CPU.  Once https://github.com/pandas-dev/pandas/issues/43881 is resolved and Dask has a native multi-col sort we can probably remove them entirely.  @charlesbluca is correct that I was originally intending to match the the case when native mult-col sorting is not supported.  

I think it's ok to safely remove persist in the initial try state and return the dataframe directly